Caboodle/Email/Email.android.cs

```diff
@@ -0,0 +1,154 @@
+﻿using System;
+using System.Collections.Generic;
+using System.Linq;
+using System.Threading.Tasks;
+using Android;
+using Android.Content;
+using Android.OS;
+using Android.Text;
+using Android.Net;
+
+namespace Microsoft.Caboodle
+{
+    public static partial class Email
+    {
+        /// Indicating whether Compose Dialog is available
+        public static bool IsComposeSupported
+            => Platform.IsIntentSupported(CreateIntent(
+                new EmailMessage()
+                {
+                    RecipientsTo = new string[] {"mailto:caboodle@mailinator.com" },
+                }));
+
+        /// Indicating whether Sending in background is available
+        public static bool IsSendSupported
+        {
+            get;
+        }
+
+        private static Intent CreateIntent(EmailMessage message)
+        {
+            if (message == null)
+                throw new ArgumentNullException(nameof(message));
+            message.Validate();
+
+            return CreateIntent(message.RecipientsTo, message.RecipientsCC, message.RecipientsBCC, message.Subject, message.Body);
+        }
+
+        private static Intent CreateIntent(
+            string[] recipientsto,
+            string[] recipientscc,
+            string[] recipientsbcc,
+            string subject = null,
+            string body = null,
+            string bodymimetype = null,
+            string[] attachmentspaths = null
+            )
+        {
+            var uri = Android.Net.Uri.Parse("mailto:");
+
+            // TODO: Tuning for Intent Choosers/Pickers
+            // Numerous apps can respond to ActionSend
+            var i = new Intent(Intent.ActionSend, uri);
+            i.SetData(uri);
+            i.SetType("plain/text");    // i.SetType("message/rfc822");
+
+            if (! (recipientsto?.Length > 0))
+                i.PutExtra(Intent.ExtraEmail, recipientsto);
+
+            if (!(recipientsto?.Length > 0))
+                i.PutExtra(Intent.ExtraCc, recipientsto);
+
+            if (!(recipientsto?.Length > 0))
+                i.PutExtra(Intent.ExtraBcc, recipientsto);
+
+            if (!string.IsNullOrWhiteSpace(subject))
+                i.PutExtra(Intent.ExtraSubject, subject);
+
+            if (!string.IsNullOrWhiteSpace(body))
+            {
+                if (bodymimetype == "text/plain")
+                {
+                    i.PutExtra(Intent.ExtraText, body);
+                }
+                else if (bodymimetype == "text/html")
+                {
+                    i.PutExtra(Intent.ExtraHtmlText, Html.FromHtml(body, FromHtmlOptions.ModeLegacy));
+                }
+                else
+                {
+                    throw new EmailException($"Invalid {nameof(bodymimetype)}: {bodymimetype}");
```

Comment:
This should not throw, fall back to plain test if it is not html as there are only 2 formats: HTML and Plain